### ISSUE TITLE
Snap Confinement prevents access to `/boot`

### ISSUE THREAD

**Issue by jnsgruk** [parca-dev/parca-agent#692]
The recent refactoring in #663 introduced more rigorous checks for whether eBPF is adequately configured on the host before profiling begins.

One of the places that can be checked is in `/boot/config-$(uname- r)`. This is the default location for the kernel config on Ubuntu and other distros. Those distributions that keep it in `/proc/config.gz` are supported currently by the `system-observe` interface.

As far as I can tell, there isn't a [supported interface](https://snapcraft.io/docs/supported-interfaces) that gets access to `/boot`. I'm talking with the snapd maintainers to get this sorted, and will post updates here as I go

**Comment by jnsgruk** [comment]
Started a discussion on [snapcraft.io[(https://forum.snapcraft.io/t/new-interface-proposal-kernel-config/31253) following some internal discussions.

**Comment by jnsgruk** [comment]
After a little discussion, it seems we might be able to get a change into the existing `system-observe` interface, which means a very easy ride for the snap once the changes land in snapd. I've started [a PR](https://github.com/snapcore/snapd/pull/12030) on snapd, but I expect it'll go through a couple of rounds of changes/reviews.

**Comment by jnsgruk** [comment]
Changes are now merged into snapd, I'll get a PR ready for when they land in the next release.

**Comment by jnsgruk** [comment]
Okay - done some testing and we're not quite there. There is a detail surrounding how to get access to `/boot` inside the snap, as by default the `/boot` dir is taken from the `core` snap, not the host system. I've drafted another patch for snapd that resolves the issue, and makes Parca Agent start just fine again. Working with the snapd team to get it landed.

**Comment by jnsgruk** [comment]
Once [this PR](https://github.com/snapcore/snapd/pull/12061) lands, we should be back in business.

**Comment by jnsgruk** [comment]
This is now resolved, the snap is now using classic confinement, and that ability is live in the store. Sorry for the delay!
